pydantic/main.py

```diff
@@ -384,7 +384,7 @@ def __setattr__(self, name, value):  # noqa: C901 (ignore complexity)
 
             known_field = self.__fields__.get(name, None)
             if known_field:
-                original_value = self.__dict__.pop(name)
+                original_value = self.__dict__[name]
```

Comment:
We used to check `self.dict(exclude={name})` [before](https://github.com/samuelcolvin/pydantic/pull/2000/files#diff-8708af8d26c26644976866f974e0b810b0940ede43ff487e457057d5a1f07999L370) and I'm the one who suggested to just pop the value and use `self.__dict__` to be faster. I think we may have some undesired behaviour if we keep the whole dictionary since `values` will have the previous value and `v` will have the new one.
I'm trying to think on possible edge cases with validators that would use `values`.
If there are such cases we could use `self.dict(exclude={name})` or use a `try...except...`

Reply:
Ah, I see, that makes sense.